pkg/R/sparkR.R

```diff
@@ -121,45 +119,46 @@ sparkR.init <- function(
   if (sparkRExistingPort != "") {
     sparkRBackendPort <- sparkRExistingPort
   } else {
+    path <- tempfile(pattern = "backend_port")
     if (Sys.getenv("SPARKR_USE_SPARK_SUBMIT", "") == "") {
       launchBackend(classPath = cp,
                     mainClass = "edu.berkeley.cs.amplab.sparkr.SparkRBackend",
-                    args = as.character(sparkRBackendPort),
+                    args = path,
                     javaOpts = paste("-Xmx", sparkMem, sep = ""))
     } else {
       # TODO: We should deprecate sparkJars and ask users to add it to the
       # command line (using --jars) which is picked up by SparkSubmit
       launchBackendSparkSubmit(
           mainClass = "edu.berkeley.cs.amplab.sparkr.SparkRBackend",
-          args = as.character(sparkRBackendPort),
+          args = path,
           appJar = .sparkREnv$assemblyJarPath,
           sparkHome = sparkHome,
           sparkSubmitOpts = Sys.getenv("SPARKR_SUBMIT_ARGS", ""))
     }
+    for (i in 1:100) {
+      if (file.exists(path)) {
+        break
+      }
+      Sys.sleep(0.1)
```

Comment:
Yeah but I am more worried about the cases where it takes longer than 10 seconds. Sometimes in the travis build and also in low memory containers I've seen longer delays. So I'd like to wait up to 1 minute in total and in that case checking every 0.1 seconds seems very wasteful. 

We could use a lower factor than 2, say 1.5 -- In that case if we use 13 iterations of the loop the times at which we check would be
```
0.150  0.375  0.713  1.219  1.978  3.117  4.826  7.389 11.233 17.000 25.649 38.624 58.086
```

So we'd check at 0.7, 1, 2, 3 seconds (covers common cases) but also handle cases up to 1 minute long.